### ISSUE TITLE
Program does not stop when windows terminal is closed.

### ISSUE THREAD

**Issue by kotenok2000** [microsoft/terminal#13586]
### Windows Terminal version

1.14.1963.0

### Windows build number

10.0.22000.795

### Other Software

doecm  1.10
https://mklasson.com/factors/index.php
https://mklasson.com/doecm_110_win.zip

### Steps to reproduce

Unpack doecm_110_win.zip 
run run.bat
close windows terminal window
ecm.exe gets killed. Doecm process doesn't get killed and tries to spawn ecm.exe 
each time doecm tries to spawn  ecm.exe cmd.exe fails with 0xc0000142 error

### Actual Behavior

When i close windows terminal window ecm.exe gets killed, but doecm.exe process doesn't get killed and tries to spawn ecm.exe 
each time doecm tries to spawn  ecm.exe cmd.exe fails with 0xc0000142 error

**Comment by DHowett** [comment]
Does this happen outside of Windows Terminal?

**Comment by kotenok2000** [comment]
It doesn't happen when i run it in conhost.

**Comment by zadjii-msft** [comment]
`run.bat`

```bat
start /B /LOW /WAIT doecm.exe %1 %2 %3 %4 %5 %6 %7 %8 %9
```

There's a `int main( int argc, char ** argv ) {` in the doecm.cc file, which makes me think this is a console subsystem application, not a windows subsystem one. 

From the included README: 

> One quirk of the "start" program used to set priority in run.bat is that it 
> disables ctrl-c. Ctrl-break still works though.


`start /?`

```
Starts a separate window to run a specified program or command.

START ["title"] [/D path] [/I] [/MIN] [/MAX] [/SEPARATE | /SHARED]
      [/LOW | /NORMAL | /HIGH | /REALTIME | /ABOVENORMAL | /BELOWNORMAL]
      [/NODE <NUMA node>] [/AFFINITY <hex affinity mask>] [/WAIT] [/B]
      [/MACHINE <x86|amd64|arm|arm64>][command/program] [parameters]
...
    B           Start application without creating a new window. The
                application has ^C handling ignored. Unless the application
                enables ^C processing, ^Break is the only way to interrupt
                the application.
```

So, doecm.exe gets a copy of the current console handles, not a new set of handles. 

Unfortunately, there's not a lot of other source there to go off of. Seems like the Terminal is closing out the console handle, but for whatever reason, doecm.exe doesn't have anything to check if the console handle is still open and close itself. So it goes on, tries to spawn an `ecm.exe`, and that can't inherit the console (because there is no console), and that throws the 0xc0000142. That's mostly conjecture, based off the little bit of source code included.  

We may need the maintainer to hop in here to help ID what's going on...

**Comment by kotenok2000** [comment]
Bump.
I think the bot is buggy. it says that the issue "Needs-Author-Feedback" but in actuality it "Needs-Attention"

**Comment by zadjii-msft** [comment]
You know, yea that's a bit of a bug. This: ![image](https://user-images.githubusercontent.com/18356694/184027476-265cfa6a-bc0b-4028-a50a-74925df24ece.png) triggered the bot's "we need the author's feedback" routine accidentally.

That being said, we still do need direct insight from the maintainer of `doecm` here. From what I can see of the source, I would expect closing the terminal (and thereby closing the console) to close this process. It kinda feels like something that needs to be reported to doecm.

I'll give the bot another week, but there's not much more we can investigate here...

**Comment by kotenok2000** [comment]
why does conhost closes the process, but windows terminal doesn't?

**Comment by lhecker** [comment]
This was fixed by #14282.
